docs/src/references/mappers/c8y-mapper.md

```diff
@@ -1325,3 +1325,230 @@ te/device/main///cmd/log_upload/<cmd_id>
+
+#### Phase 1
+
+**Live mapper**
+
+When the mapper is live, if telemetry messages for a new entity is received before its registration message,
+they are simply dropped until that registration message is processed, with the assumption that,
+telemetry data updates will repeat eventually and hence will make up for the lost ones.
+But, all non-telemetry data like twin updates, capability messages etc are cached in an in-memory "unregistered entity store",
+and will be registered in the entity store once the registration message eventually arrives.
+
+Any nested entity registration messages that arrives before their ancestors are registered
+will also be kept in the same "unregistered entity store" until all their ancestors are registered.
+
+To minimize dropping a lot of telemetry messages, they can also be cached briefly in a bounded queue,
+as in most practical cases, even if the registration messages are late, they will be delivered
+within a fair amount of time and we can avoid dropping messages in such cases.
+Keeping that queue bounded is critical to prevent it from growing indefinitely.
+
+There will also be max time limit for an unregistered entity to stay cached, after which it will be dropped.
+
+**On mapper restart**
+
+Sticking to the same solution of dropping telemetry messages on a mapper restart might result in loosing too much data,
+when the volume of queued telemetry messages are very high on the broker,
+increasing the probability of a lot of those telemetry messages getting delivered to the mapper first,
+before their respective registration messages are.
+
+If all the registered entities are persisted on disk and kept up-to-date while live,
+the entity store can easily be rebuilt on startup before processing any new messages.
+This will eliminate any possibility of message races for all previously registered entities.
+
+There could still be new entities for which the registration messages and other telemetry messages
+were published while the mapper was down.
+On startup, a message delivery race for such entities can happen as well,
+but they are handled the same way such messages are handled by the live mapper:
+All telemetry messages would simply be dropped until the registration message for that entity arrives.
+
+The entity store could be rebuilt using the retained messages on the MQTT broker as well,
+but it is not easy to distinguish the previously registered entities from any new registration messages
+that may have arrived while the mapper was down.
+This problem doesn't exist while using a persisted entity store, which only contains previously registered entities.
+This can also be used to easily distinguish new registrations from the previously registered ones.
+
+:::note
+Since the real entity store is persisted, the "unregistered entity store" can also be persisted on disk
+to avoid memory pressure caused by rogue clients pounding data without ever registering themselves.
```

Comment:
I would not bother with that, focusing on well-behaving clients and dropping the oldest messages if the in-memory cache starts to be full.